opensrp-opd/src/main/java/org/smartregister/opd/interactor/BaseOpdRegisterActivityInteractor.java

```diff
@@ -51,6 +60,44 @@ public void saveRegistration(List<OpdEventClient> opdEventClientList, String jso
 
     }
 
+    @Override
+    public void saveEvent(@NonNull final Event event, @NonNull final OpdRegisterActivityContract.InteractorCallBack callBack) {
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                saveEventInDb(event);
+                appExecutors.mainThread().execute(new Runnable() {
+                    @Override
+                    public void run() {
+                        callBack.onEventSaved();
+                    }
+                });
+            }
+        };
+
+        appExecutors.diskIO().execute(runnable);
+    }
+
+
+    private void saveEventInDb(@NonNull Event event) {
+        try {
+            CoreLibrary.getInstance().context().getEventClientRepository().addEvent(event.getBaseEntityId()
+                    , new JSONObject(JsonFormUtils.gson.toJson(event)));
+        } catch (JSONException e) {
+            Timber.e(e);
+        }
+
+        long lastSyncTimeStamp = getAllSharedPreferences().fetchLastUpdatedAtDate(0);
+        Date lastSyncDate = new Date(lastSyncTimeStamp);
+
+        try {
+            getClientProcessorForJava().processClient(getSyncHelper().getEvents(lastSyncDate, BaseRepository.TYPE_Unprocessed));
+            getAllSharedPreferences().saveLastUpdatedAtDate(lastSyncDate.getTime());
+        } catch (Exception e) {
+            Timber.e(e);
+        }
```

Comment:
This can be in its own function, or what do you think?

Reply:
It kinda already is, if the lines are removed to another method. It leaves this method with about 5 lines.

Reply:
Will be be cleaner that way, cause that method does more than save Event in db

Reply:
Fixed